pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -109,39 +115,46 @@ let
             ( bin.${pname} // { tlType = "bin"; } );
     };
 
-  # for daily snapshots
-  snapshot = {
-    year = "2022";
-    month = "12";
-    day = "27";
+  version = {
+    # day of the snapshot being taken
+    year = "2023";
+    month = "03";
+    day = "19";
+    # TeX Live version
+    texliveYear = 2022;
+    # final (historic) release or snapshot
+    final = true;
   };
 
   # The tarballs on CTAN mirrors for the current release are constantly
   # receiving updates, so we can't use those directly. Stable snapshots
   # need to be used instead. Ideally, for the release branches of NixOS we
   # should be switching to the tlnet-final versions
   # (https://tug.org/historic/).
-  urlPrefixes = [
-    # tlnet-final snapshot
-    "http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/${bin.texliveYear}/tlnet-final/archive"
-    "ftp://tug.org/texlive/historic/${bin.texliveYear}/tlnet-final/archive"
-
-    # Daily snapshots hosted by one of the texlive release managers
-    #"https://texlive.info/tlnet-archive/${snapshot.year}/${snapshot.month}/${snapshot.day}/tlnet/archive"
+  urlPrefixes = with version; lib.optionals final  [
+    # tlnet-final snapshot; used when texlive.tlpdb is frozen
+    # the TeX Live yearly freeze typically happens in mid-March
+    "http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/${toString texliveYear}/tlnet-final"
+    "ftp://tug.org/texlive/historic/${toString texliveYear}/tlnet-final"
+  ] ++ [
+    # daily snapshots hosted by one of the texlive release managers;
+    # used for non-final snapshots and as fallback for final snapshots that have not reached yet the historic mirrors
+    # please note that this server is not meant for large scale deployment and should be avoided on release branches
+    # https://tug.org/pipermail/tex-live/2019-November/044456.html
+    "https://texlive.info/tlnet-archive/${year}/${month}/${day}/tlnet"
   ];
 
-  tlpdb = fetchurl {
-    # use the same mirror(s) as urlPrefixes above
-    urls = map (up: "${up}/../tlpkg/texlive.tlpdb.xz") urlPrefixes;
+  tlpdbxz = fetchurl {
+    urls = map (up: "${up}/tlpkg/texlive.tlpdb.xz") urlPrefixes;
     hash = "sha256-vm7DmkH/h183pN+qt1p1wZ6peT2TcMk/ae0nCXsCoMw=";
   };
 
-  tlpdb-nix = runCommand "tlpdb.nix" {
-    inherit tlpdb;
+  tlpdbNix = runCommand "tlpdb.nix" {
+    inherit tlpdbxz;
```

Comment:
Before calling this fully done: how do I hide this from [search.nixos.org](https://search.nixos.org/packages?channel=unstable&query=tlpdb)? It's ugly!

Reply:
There doesn't appear to be a pretty way to hide derivations.

the best solutions I can think of are
- nesting it inside another attr-set without `recurseForDerivations` attribute (like `pkgs.texlive.updater.tlpdbNix`). The derivations for the fixed hashes from the other PR could also be inserted there.
- moving it to the tests. After all, this is the only place where it is used aside from the manual invocation when updating. But reading something like „run `cp $(nix-build ../../../.. -A tests.texlive.tlpdbNix) tlpdb.nix`“ in the update instructions might also  cause confusion.

Also possible, but imho even worse:
- hiding it as the return value of a constant function.
- hiding it in a list
- hiding it as a passthru-attribute of one of the mostly unrelated derivations
- making it completely inaccessible via the pkgs datastructure (as before).

Reply:
Ok, I have opted for nesting under `texlive.tlpdb`, and while I was there, I exposed `tlpdbxz` and `version`.